### PR TITLE
Adds different text to grant finish screen

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/4814

UI PR: https://github.com/brave/brave-ui/pull/499

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
Test plan 1:
- enable rewards
- claim grant
- make sure that text on a final screen is ok (ugp text)

Test plan 1:
- enable rewards
- restore wallet with ad grant waiting
- claim ad grant
- make sure that text on a final screen is ok (ads text)

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json [hunk 1]
```diff
@@ -438,5 +430,29 @@
...
+  "grantFinishTokenTitleUGP": {
+    "message": "Free Token Grant",
+    "description": "Token title on success screen"
+  },
+  "grantFinishTitleAds": {
+    "message": "Brave Ads Rewards!",
+    "description": "Cosmetic text upon captcha success for ads"
+  },
+  "grantFinishTextAds": {
+    "message": "Your rewards grant from Brave Ads is on its way",
```
Comment (tmancey) [inline components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json]: needs a full-stop to be consistent with UGP

File: components/brave_rewards/resources/extension/brave_rewards/utils.ts [hunk 2]
```diff
@@ -48,8 +49,19 @@ export const getGrants = (grants?: RewardsExtension.Grant[]) => {
...
+    return grant
+  }
+
+  grant.finishTitle = getMessage('grantFinishTitleUGP')
+  grant.finishText = getMessage('grantFinishTextUGP')
+  grant.finishTokenTitle = getMessage('grantFinishTokenTitleUGP')
+
+  if (grant.type === 'ads') {
     grant.expiryTime = 0
+    grant.finishText = getMessage('grantFinishTextAds')
```
Comment (tmancey) [inline components/brave_rewards/resources/extension/brave_rewards/utils.ts]: nit: for consistency swap text/title lines around so they match above

File: components/brave_rewards/resources/ui/components/grant.tsx [hunk 3]
```diff
@@ -135,14 +135,43 @@ class Grant extends React.Component<Props, State> {
     )
   }
 
+  grantFinish = (type: string, tokens: string, date: string) => {
+    let title = getLocale('grantFinishTitleUGP')
```
Comment (tmancey) [inline components/brave_rewards/resources/ui/components/grant.tsx]: nit: could we not move to these to an `else` statement

File: components/brave_rewards/resources/extension/brave_rewards/utils.ts [hunk 4]
```diff
@@ -48,8 +49,19 @@ export const getGrants = (grants?: RewardsExtension.Grant[]) => {
 }
 
 export const getGrant = (grant?: RewardsExtension.GrantInfo) => {
-  if (grant && grant.type === 'ads') {
+  if (!grant) {
+    return grant
+  }
+
+  grant.finishTitle = getMessage('grantFinishTitleUGP')
```
Comment (tmancey) [inline components/brave_rewards/resources/extension/brave_rewards/utils.ts]: nit: could we not move to these to an `else` statement

File: components/resources/brave_components_strings.grd [hunk 5]
```diff
@@ -524,6 +522,13 @@
       <message name="IDS_BRAVE_UI_SHOW_ALL" desc="">Show All</message>
       <message name="IDS_BRAVE_UI_SUPPORTED_SITES" desc="">Supported Sites</message>
 
+      <message name="IDS_BRAVE_UI_GRANT_FINISH_TOKEN_ADS" desc="Token text on final grant screen for UGP">Your Brave Ads Token Grant</message>
+      <message name="IDS_BRAVE_UI_GRANT_FINISH_TEXT_ADS" desc="Text on final grant screen for UGP">Your rewards grant from Brave Ads is on its way</message>
```
Comment (tmancey) [inline components/resources/brave_components_strings.grd]: needs a full-stop to be consistent with UGP

File: components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json (hunk 1 above)
Comment (NejcZdovc) [reply components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json]: done

File: components/brave_rewards/resources/extension/brave_rewards/utils.ts (hunk 2 above)
Comment (NejcZdovc) [reply components/brave_rewards/resources/extension/brave_rewards/utils.ts]: done

File: components/resources/brave_components_strings.grd (hunk 5 above)
Comment (NejcZdovc) [reply components/resources/brave_components_strings.grd]: done

**Pull request merged**